src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderService.java

```diff
@@ -9,35 +11,125 @@
       UUID id, SnapshotAccessRequest snapshotAccessRequest, String email) {
     return snapshotRequestDao.create(id, snapshotAccessRequest, email);
   }
 
-  public SnapshotBuilderGetConceptsResponse getConceptChildren(UUID datasetId, Integer conceptId) {
+  // TODO - replace manual query with query builder
+  // The query builder will handle switching between cloud platforms, so we won't
+  // have this duplicated logic in the final solution.
+  private String buildConceptChildrenQuery(
+      Dataset dataset, int conceptId, AuthenticatedUserRequest userRequest) {
+    var cloudPlatformWrapper = CloudPlatformWrapper.of(dataset.getCloudPlatform());
+    if (cloudPlatformWrapper.isGcp()) {
+      var bqTablePrefix =
+          String.format(
+              "%s.%s",
+              dataset.getProjectResource().getGoogleProjectId(),
+              BigQueryPdao.prefixName(dataset.getName()));
+
+      return String.format(
+          """
+                  SELECT c.concept_name, c.concept_id FROM `%s.concept` AS c%n\
+                  WHERE c.concept_id IN%n\
+                  (SELECT c.descendant_concept_id FROM `%s.concept_ancestor` AS c%n\
+                  WHERE c.ancestor_concept_id = %d)%n\
+          """,
+          bqTablePrefix, bqTablePrefix, conceptId);
+    } else if (cloudPlatformWrapper.isAzure()) {
+      String conceptParquetFilePath = IngestUtils.getSourceDatasetParquetFilePath("concept");
+      String conceptAncestorFilePath =
+          IngestUtils.getSourceDatasetParquetFilePath("concept_ancestor");
+
+      String datasourceName =
+          datasetService.getOrCreateExternalAzureDataSource(dataset, userRequest);
+      return String.format(
+          """
+            SELECT c.concept_name, c.concept_id FROM%n\
+            (SELECT * FROM OPENROWSET(BULK '%s', DATA_SOURCE = '%s', FORMAT = 'parquet') AS alias951024263)%n\
+            AS c WHERE c.concept_id IN%n\
+            (SELECT c.descendant_concept_id FROM%n\
+            (SELECT * FROM OPENROWSET(BULK '%s', DATA_SOURCE = '%s', FORMAT = 'parquet') AS alias625571305)%n\
+            AS c WHERE c.ancestor_concept_id = %d)%n\
+          """,
+          conceptParquetFilePath,
+          datasourceName,
+          conceptAncestorFilePath,
+          datasourceName,
+          conceptId);
+
+    } else {
+      throw new NotImplementedException("Cloud platform not implemented");
+    }
+  }
+
+  private <T> List<T> runSnapshotBuilderQuery(
+      String cloudSpecificSql,
+      Dataset dataset,
+      Function<TableResult, List<T>> gcpFormatQueryFunction,
+      Function<ResultSet, T> synapseFormatQueryFunction) {
+    var cloudPlatformWrapper = CloudPlatformWrapper.of(dataset.getCloudPlatform());
+    if (cloudPlatformWrapper.isGcp()) {
+      return bigQueryDatasetPdao.runQuery(cloudSpecificSql, dataset, gcpFormatQueryFunction);
+    } else if (cloudPlatformWrapper.isAzure()) {
+      return azureSynapsePdao.runQuery(cloudSpecificSql, synapseFormatQueryFunction);
```

Comment:
It's a minor thing but since you're checking platform here it's probably better to reference the methods directly here instead of passing them in.
```suggestion
      return bigQueryDatasetPdao.runQuery(cloudSpecificSql, dataset, AggregateBQQueryResultsUtils::aggregateConceptResults);
    } else if (cloudPlatformWrapper.isAzure()) {
      return azureSynapsePdao.runQuery(cloudSpecificSql, AggregateSynapseQueryResultsUtils::aggregateConceptResult);
```

Reply:
My thought is that we'll reuse this runSnapshotBuilderQuery() method, so we'll want to pass in the correct aggregate methods per call.